src/pyqasm/modules/base.py

```diff
@@ -156,6 +166,7 @@ def remove_measurements(self, in_place: bool = True) -> Optional["QasmModule"]:
         Returns:
             QasmModule: The module with the measurements removed if in_place is False
         """
+        self._user_operations.append("remove_measurements")
```

Comment:
While this works, a cleaner approach would be to use decorators in Python. You can define a decorator which can be called whenever a function is executed and then append the name to the `user_operation` list , something like - 

```python
def track_user_operation(func):
    def wrapper(self, *args, **kwargs):
        self._user_operations.append(func.__name__)
        return func(self, *args, **kwargs)
    return wrapper

class QasmModule(ABC):
    # ... existing code ...

    @track_user_operation
    def validate(self):
        # ... method body ...
```